pkg/kustomize/kustomize_cli_test.go

```diff
@@ -0,0 +1,63 @@
+package kustomize_test
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/kustomize"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+)
+
+func TestKustomizeCLI_FindKustomize_HasKustomize(t *testing.T) {
+	origJXHome, testJXHome, err := testhelpers.CreateTestJxHomeDir()
+	if err != nil {
+		t.Errorf("failed to create a test JX Home directory")
+	}
+
+	defer func() {
+		err := testhelpers.CleanupTestJxHomeDir(origJXHome, testJXHome)
+		if err != nil {
+			log.Logger().Warnf("unable to remove temporary directory %s: %s", testJXHome, err)
+		}
+	}()
+
+	err = os.MkdirAll(filepath.Join(testJXHome, "bin/base/charts"), os.ModePerm)
+	err = os.MkdirAll(filepath.Join(testJXHome, "bin/base/templates"), os.ModePerm)
+	err = os.MkdirAll(filepath.Join(testJXHome, "bin/staging"), os.ModePerm)
+	_, err = os.Create(filepath.Join(testJXHome, "bin/base/Charts.yaml"))
+	_, err = os.Create(filepath.Join(testJXHome, "bin/base/values.yaml"))
+	_, err = os.Create(filepath.Join(testJXHome, "bin/base/kustomization.yaml"))
+	_, err = os.Create(filepath.Join(testJXHome, "bin/staging/values.yaml"))
+	_, err = os.Create(filepath.Join(testJXHome, "bin/staging/kustomization.yaml"))
+	_, err = os.Create(filepath.Join(testJXHome, "bin/base/charts/kustomization.yaml"))
+	if err != nil {
```

Comment:
hmm, technically you only check the last os.Create call here on error. You need to check all.
Another maybe easier approach is to create this directory structure in _testdata_ and just run the test against this dir. You can have multiple directory structures/setups under _testdata_ running several tests. Easier than creating the structure on the fly.

Look in the source tree for _testdata_ directories and see how it is done there. _testdata_ directories are treated special by Go and won't be compiled or otherwise added to the final binary.